pkg/secrets/providers.go

```diff
@@ -0,0 +1,71 @@
+package secrets
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/klog/v2"
+
+	"kusionstack.io/kusion/pkg/apis/secrets"
+)
+
+type Providers struct {
+	lock     sync.RWMutex
```

Comment:
Do we need a thread-safe map here?  Maybe the [syncMap](https://pkg.go.dev/sync#Map) in the standard library is helpful.